src/components/SearchBar/SearchBar.test.tsx

```diff
@@ -0,0 +1,43 @@
+import { fireEvent, render, screen, waitFor } from '@testing-library/react';
+import SearchBar from './SearchBar';
+
+describe('SearchBar', () => {
+  const changeStub = vi.fn();
+  const storage: Record<string, string> = {};
+
+  beforeAll(() => {
+    vi.spyOn(window, 'fetch').mockImplementation(() => {
+      return Promise.resolve({
+        json: () => Promise.resolve({ data: '111' }),
+      } as Response);
+    });
+    Object.defineProperty(window, 'localStorage', {
+      value: {
+        setItem: vi.fn((key: string, value: string) => {
+          storage[key] = value;
+        }),
+        getItem: vi.fn((key: string) => storage[key]),
+      },
+    });
+  });
+
+  beforeEach(() => {
+    vi.clearAllMocks();
+  });
+
+  it('should render search bar', () => {
+    render(<SearchBar valChange={changeStub}></SearchBar>);
```

Comment:
also suggest to wrap all render calls and user interactions (fireEvent.change and etc.) to `act(() => ...)`
https://testing-library.com/docs/react-testing-library/api/#act
_This makes your test run closer to how React works in the browser._